File: IdeaProject/КИТПО/kitpo_lr_3-2/src/main/scala/MyTreeFactory.scala
class MyTreeFactory {
  def createTree(_type: String): MyTree =
  {
    var list: MyTree = null;

    _type match
    {
      case "String" =>
      {
        list = new MyTreeStr(_type)
      }
      case "Integer" =>
      {
        //list = new MyTreeInt(_type)
      }
    }
    return list
  }
}

File: IdeaProject/КИТПО/kitpo_lr_1-2/src/MyTreeFactory.java
public class MyTreeFactory {

    public MyTree createTree(String type){
        MyTree tree = null;

        switch (type)
        {
            case "String":
            {
                tree = new MyTreeStr(type);
                break;
            }
            case "Integer": {
                tree = new MyTreeInt(type);
                break;
            }
        }

        return tree;
    }
}
